src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php

```diff
@@ -0,0 +1,303 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Tests\Caster;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\VarDumper\Test\VarDumperTestTrait;
+
+class DOMCasterTest extends TestCase
+{
+    use VarDumperTestTrait;
+
+    public function testCastImplementation()
+    {
+        $implementation = new \DOMImplementation();
+
+        $this->assertDumpEquals(<<<'EODUMP'
+DOMImplementation {
+  Core: "1.0"
+  XML: "2.0"
+}
+EODUMP, $implementation);
+    }
+
+    /**
+     * @requires PHP 8.4
+     */
+    public function testCastModernImplementation()
+    {
+        $implementation = new \DOM\Implementation();
+
+        $this->assertDumpEquals(<<<'EODUMP'
+DOM\Implementation {
+  Core: "1.0"
+  XML: "2.0"
+}
+EODUMP, $implementation);
+    }
+
+    public function testCastNode()
+    {
+        $doc = new \DOMDocument();
+        $doc->loadXML('<foo><bar/></foo>');
+        $node = $doc->documentElement->firstChild;
+
+        $this->assertDumpMatchesFormat(<<<'EODUMP'
+DOMElement {%A
+  +ownerDocument: ? ?DOMDocument
+  +namespaceURI: ? ?string
+  +prefix: ? string
+  +localName: ? ?string
+%A}
+EODUMP, $node);
+    }
+
+    /**
+     * @requires PHP 8.4
+     */
+    public function testCastModernNode()
+    {
+        $doc = \DOM\XMLDocument::createFromString('<foo><bar/></foo>');
+        $node = $doc->documentElement->firstChild;
+
+        $this->assertDumpMatchesFormat(<<<'EODUMP'
+DOM\Element {%A
+  +baseURI: ? string
+  +isConnected: ? bool
+  +ownerDocument: ? ?DOM\Document
+%A}
+EODUMP, $node);
+    }
+
+    public function testCastDocument()
+    {
+        $doc = new \DOMDocument();
+        $doc->loadXML('<foo><bar/></foo>');
+
+        $this->assertDumpMatchesFormat(<<<'EODUMP'
+DOMDocument {%A
```

Comment:
let's indent them all now that we can

Reply:
Here you go. It is something that may be done in other caster tests as well?

Reply:
let's keep the git history clean :)